_GNU_SOURCE の再定義警告の解消

## 概要
CI の `Run Make test` ログで、以下の 2 点が判明

1. `_GNU_SOURCE` の二重定義による再定義警告
 - 例: `../includes/repl.h:15: warning: "_GNU_SOURCE" redefined`
 - `<command-line>`（コンパイル時の定義）とヘッダ側の定義が競合

2. valgrind の FILE DESCRIPTORS リポートにおける「親プロセス由来の FD」の扱い
 - 例:
   ```
   ==4377== FILE DESCRIPTORS: 5 open (3 std) at exit.
   ==4377== Open file descriptor 145:
   ==4377==    <inherited from parent>
   ==4377== Open file descriptor 142:
   ==4377==    <inherited from parent>
   ...
   ==4377== All heap blocks were freed -- no leaks are possible
   ```
 - これは CI 上のランナー/親プロセスから継承された FD であり、テスト対象コードのリークではない

```
#ifndef _GNU_SOURCE
# define _GNU_SOURCE
#endif
```
すでにコンパイラまたはビルドフラグで定義されている場合は再定義を避けられる

--- a/includes/repl.h
+++ b/includes/repl.h
@@ -3,16 +3,19 @@
 /*                                                        :::      ::::::::   */
 /*   repl.h                                             :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: tshimizu <tshimizu@student.42tokyo.jp>     +#+  +:+       +#+        */
+/*   By: nkojima <nkojima@student.42tokyo.jp>       +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2025/11/24 12:48:38 by tshimizu          #+#    #+#             */
-/*   Updated: 2025/12/06 16:43:08 by tshimizu         ###   ########.fr       */
+/*   Updated: 2025/12/14 17:37:10 by nkojima          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #ifndef REPL_H
 # define REPL_H
-# define _GNU_SOURCE
+
+# ifndef _GNU_SOURCE
+#  define _GNU_SOURCE
+# endif
 
 # include <signal.h>
 # include <stdbool.h>

